app/src/main/java/com/example/felipe/criminalintent/CrimeFragment.kt

```diff
@@ -0,0 +1,49 @@
+package com.example.felipe.criminalintent
+
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import kotlinx.android.synthetic.main.fragment_crime.view.*
+
+class CrimeFragment() : Fragment() {
+    private lateinit var crime: Crime
```

Comment:
I don't know, if he moves the init to the `onCreateView` (so he doesn't have to override the `onCreate` just for that), it's OK to keep it. He just needs to be aware of the lifecycle.
If he inits it right here, then yes, `lateinit` is not needed.

Reply:
Lateinits are dangerous, so I try to avoid them. I know, if you will initializate it in this method or in the onviewcreated you can use it. However, as a personal rule, if I can avoid them I do that. In this case, you can avoid it if you assign the variable in the field declaration. What do yo think?

Reply:
BTW, I think this variable is not used, yo can remove it

Reply:
They are dangerous yes but if they're handled correctly, like in this case, I don't see why you can't use them. It's true that you must tread very carefully and it's good to be aware of that.

It's better to be as defensive as possible when programming, so it's better if you init the variable here @f7deleon as @matir91 said, but just know that the use of `lateinit` was not wrong in this case.